File: Tutorial2InjectConstructor/src/main/java/com/example/tutorial2injectconstructor/MyApplication.java
package com.example.tutorial2injectconstructor;

import android.app.Application;

import com.example.tutorial2injectconstructor.di.DaggerMyComponent;
import com.example.tutorial2injectconstructor.di.MyComponent;
import com.example.tutorial2injectconstructor.di.SharedPreferencesModule;

public class MyApplication extends Application {
    private MyComponent myComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        myComponent = DaggerMyComponent.builder()
                .sharedPreferencesModule(new SharedPreferencesModule(getApplicationContext()))
                .build();

    }

    public MyComponent getMyComponent() {
        return myComponent;
    }
}

File: Tutorial2-1InjectConstructor/src/main/java/com/example/tutorial2injectconstructor/MyApplication.kt
package com.example.tutorial2injectconstructor

import android.app.Application
import com.example.tutorial2injectconstructor.di.DaggerMyComponent
import com.example.tutorial2injectconstructor.di.MyComponent
import com.example.tutorial2injectconstructor.di.SharedPreferencesModule

class MyApplication : Application() {

    lateinit var myComponent: MyComponent
        private set

    override fun onCreate() {
        super.onCreate()

        myComponent = DaggerMyComponent.builder()
                .sharedPreferencesModule(SharedPreferencesModule(applicationContext))
                .build()
    }

}